models/points.go

```diff
@@ -394,85 +397,120 @@ func ParsePointsWithPrecision(buf []byte, defaultTime time.Time, precision strin
 
 }
 
-func parsePoint(buf []byte, defaultTime time.Time, precision string) (Point, error) {
+func parsePointsAppend(points []Point, buf []byte, mm []byte, defaultTime time.Time, precision string) ([]Point, error) {
 	// scan the first block which is measurement[,tag1=value1,tag2=value=2...]
 	pos, key, err := scanKey(buf, 0)
 	if err != nil {
-		return nil, err
+		return points, err
 	}
 
 	// measurement name is required
 	if len(key) == 0 {
-		return nil, fmt.Errorf("missing measurement")
+		return points, errMissingMeasurement
 	}
 
 	if len(key) > MaxKeyLength {
-		return nil, fmt.Errorf("max key length exceeded: %v > %v", len(key), MaxKeyLength)
+		return points, fmt.Errorf("max key length exceeded: %v > %v", len(key), MaxKeyLength)
 	}
 
-	// scan the second block is which is field1=value1[,field2=value2,...]
-	pos, fields, err := scanFields(buf, pos)
-	if err != nil {
-		return nil, err
-	}
+	// Since the measurement is converted to a tag and measurements & tags have
```

Comment:
This is the sort of stuff I'm worried about 😄 We're adding new escaping. Currently in 2.0 if you write 

```
influx write --bucket-id 03554e2537ef2000 'ab\=c v=1'
```

Then in TSM we will store the following series key:


```
<orgbucket>,_f=v,_m=ab\=c#!~#v
```

Now, with this PR, we would instead store:

```
<orgbucket>,_f=v,_m=ab\\=c#!~#v
```

I'm concerned that we're changing the on disk representation of the series keys.

Reply:
I can confirm though that this does change behavior for writes vs master. Issuing writes of the form 

```
$ ~/go/bin/influx write -b test 'abc v="<branch>"'
$ ~/go/bin/influx write -b test 'a=bc v="<branch>"'
$ ~/go/bin/influx write -b test 'a\=bc v="<branch>"'
$ ~/go/bin/influx write -b test 'a\\=bc v="<branch>"'
```

for both master and this pr (rebased on master), the reads come out as

```
> from(bucket:"test") |> range(start:-1d) |> keep(columns:["_measurement", "_value"])
Result: _result
Table: keys: [_measurement]
   _measurement:string           _value:string
----------------------  ----------------------
                  a=bc                      pr
                  a=bc                  master
                  a=bc                  master
Table: keys: [_measurement]
   _measurement:string           _value:string
----------------------  ----------------------
                 a\=bc                      pr
                 a\=bc                  master
Table: keys: [_measurement]
   _measurement:string           _value:string
----------------------  ----------------------
                a\\=bc                      pr
Table: keys: [_measurement]
   _measurement:string           _value:string
----------------------  ----------------------
                   abc                      pr
                   abc                  master
```

which actually is better. At least the PR writes to four distinct measurements.

The TSM index for the filr confirms the series keys:

```
<orgbucket>,_f=v,_m=a\=bc
<orgbucket>,_f=v,_m=a\\=bc
<orgbucket>,_f=v,_m=a\\\=bc
<orgbucket>,_f=v,_m=abc
```

Reply:
@zeebo but I think the measurement name you would give to InfluxQL and to `flux` would be different wouldn't it? 

That means that the transpiler won't work for those names.